.github/workflows/scripts/slack_notifications/gem_notifier.rb

```diff
@@ -0,0 +1,83 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'time'
+require 'httparty'
+require_relative 'slack_notifier'
+
+class GemNotifier < SlackNotifier
+  def self.check_for_updates(watched_gems)
+    return if gem_list_empty(watched_gems)
+    watched_gems.each do |gem_name|
+      verify_gem(gem_name) ? gem_info = verify_gem(gem_name) : return
```

Comment:
This is not related to the PR, but we're going to RubyGems twice here for each gem for the very same information.

In future we should refactor to do something like this:

```diff
- verify_gem(gem_name) ? gem_info = verify_gem(gem_name) : return
+ gem_info = verify_gem(gem_name)
+ abort("Failed to obtain info for gem '#{gem_name}'") unless gem_info
+ 
versions = gem_versions(gem_info)
```

Reply:
I'm very glad you brought this up—definitely no need to make duplicate calls for the same thing. [Addressed here](https://github.com/newrelic/newrelic-ruby-agent/pull/1432/commits/6b044062a798dca30a0c6649b73684c31246f934).